lib/internal/util.js

```diff
@@ -257,6 +257,18 @@ function promisify(original) {
 
 promisify.custom = kCustomPromisifiedSymbol;
 
+// The build in join is relatively slow (currently v8 6.0)
+function join(output, separator) {
+  var str = '';
```

Comment:
Any specific reason to use `var` instead of `let` in the whole PR?

Reply:
I did not measure `let` and `var` recently but in general using let was slower then var. I have to check at some point if that is still the case. I would rather not change it right to `let` in this PR though.